http/index.ts

```diff
@@ -15,6 +20,54 @@ import { i18n } from '../utils/lang';
 
 const i18nKey = 'http.index';
 
+export class HubSpotHttpError<T = unknown, D = unknown> extends Error {
```

Comment:
Could see this making sense having its own file (theres a precedent with `HubspotAuthError`)

Reply:
Good call, I'll move it over into it's own thing.